Lots of weapons and other things are now varying degrees more or less lethal + New gun sounds and sprites. Pulse rifle nerfs.

chaoko99 opened the pull request:
🆑 
tweak: Claymores do 10% more brute.
tweak: Replica katanas deal less damage (Used to be as deadly as claymores.)
tweak: Spears do more damage. Much stronger in one hand now. Can be thrown twice as fast.
tweak: Energy swords deal 50% more damage. 20% more likely to block attacks.
tweak: Bear traps now do slightly less than twice as much damage.
tweak: Null-Rod isn't ridiculously deadly anymore. Thank God you never found out
tweak: Pulse weapons have been nerfed. They are less accurate, take longer to shoot, and have a much higher chance to miss due to the usage of more, much weaker projectiles. They lock the wielder in place for a significant period now. Now use unique sprites for their projectiles, shooting pulses rather than lasers. No longer hitscan.
tweak: Uzi has had all values for unwielded inaccuracy shifted downwards by 1.
tweak: Tasers are no longer hitscan weapons. Shock beam is unaffected.
tweak: shock beam has a unique sound.
tweak: Energy Crossbow has some armor penetration and a minor EMP.
tweak: Phoron pistol does 50% as much damage, and uses its own sound.
soundadd: Lasers now hiss when they hit walls. New E-Carbine sounds.
/🆑

BlueNexus commented on code/modules/projectiles/guns/energy/pulse.dm:
```diff
@@ -41,7 +55,11 @@
 	wielded_item_state = null
+=======
+	accuracy = -1
+>>>>>>> fde3058... GUNS
```
You, uh. Forgot something.

chaoko99 replied:
Shitshit how did that even happen

chaoko99 replied:
Resolved.

BlueNexus commented on code/modules/projectiles/guns/energy/stun.dm:
```diff
@@ -4,10 +4,10 @@
 
 	firemodes = list(
-		list(mode_name="stun", projectile_type=/obj/item/projectile/beam/stun),
+		list(mode_name="stun", projectile_type=/obj/item/projectile/energy/electrode/taser),
```
Thoroughly against this. The tradeoff between the stun revolver and tasers is that one has a projectile but is stronger, and the other has a beam, can go through things, and is weaker. What's your reasoning for this change?

chaoko99 replied:
Because an antag cannot react to a hitscan projectile. They get hit once? They're fucked. Most antag armor has very minor resistance to tasers anyways... Tasermeming is just annoying and frustrating. If you can think of a better alternative, do tell.

Sunbeamstress replied:
I guess if you suck at SS13 gunfighting? Be aware of your surroundings. Avoid lots of windows or open areas. Pull a closet in front of you. It's so easy not to get tased if you're planning for it.

chaoko99 replied:
Based on anecdotes. Played a round on another server where all tasers used electrodes instead of hitscan projectiles. It was a lot tenser, because you could weave out of the way of the projectiles. Perhaps, maybe, a longer firing delay would make this tension exist... A big part of it was the fact that you could weave out of the way of said projectiles, rather than having to predict when it happens.

comma commented on code/modules/projectiles/guns/energy/stun.dm:
```diff
@@ -4,10 +4,10 @@
 	item_state = null	//so the human update icon uses the icon_state instead.
 	max_shots = 5
-	projectile_type = /obj/item/projectile/beam/stun
+	projectile_type = /obj/item/projectile/energy/electrode/taser
```
Why?

chaoko99 replied:
See above.

PurpleMartinJCK commented:
Needs a changelog

PurpleMartinJCK commented on code/game/objects/items/weapons/melee/energy.dm:
```diff
@@ -143,7 +143,7 @@
 
 /obj/item/weapon/melee/energy/sword/handle_shield(mob/user, var/damage, atom/damage_source = null, mob/attacker = null, var/def_zone = null, var/attack_text = "the attack")
-	if(active && default_parry_check(user, attacker, damage_source) && prob(50))
+	if(active && default_parry_check(user, attacker, damage_source) && prob(60))
```
This should take the species tag NO_BLOCK into account

chaoko99 replied:
Would it not be smarter to have that handled in the parry check itself?

chaoko99 replied:
Done. I meant the base proc, that is.

PurpleMartinJCK commented on code/modules/projectiles/guns/energy/pulse.dm:
```diff
@@ -6,15 +6,23 @@
-	max_shots = 36
+	max_shots = 108
 	w_class = ITEM_SIZE_HUGE
+<<<<<<< HEAD
```
Merge stuff needs to be fixed

chaoko99 replied:
Resolved.

PurpleMartinJCK commented on code/modules/projectiles/guns/energy/pulse.dm:
```diff
@@ -23,11 +31,17 @@
-	max_shots = 24
+	max_shots = 72
 	w_class = ITEM_SIZE_LARGE
+<<<<<<< HEAD
```
More merge stuff

chaoko99 replied:
Resolved.

Kelenius commented:
This is weird and out of the blue. What's the reasoning behind these changes?

chaoko99 commented:
Because nobody else felt like adjusting things?

Also because this is bundled in with a lot of other things. Like sound changes and new sounds.

Kelenius commented:
Are you just adjusting for the sake of adjusting? Give a clear reasoning why you think that these specific numbers should be adjusted this way.

chaoko99 commented:
>Are you just adjusting for the sake of adjusting? Give a clear reasoning why you think that these specific numbers should be adjusted this way.

>tweak: Claymores do 10% more brute.

They did exactly as much damage as Katanas, and I thought that was kinda stupid and nonsensical.

>tweak: Replica katanas deal 50% less damage.

They're fucking replicas. They shouldn't hurt much at all.

>tweak: Spears do 50% more damage. Approximately 360% more powerful in one hand. Can be thrown twice as fast.

Spears generally are meant to be used in conjunction with shields. I figured their one-hand penalty was too high.

>tweak: Energy swords deal 50% more damage. 20% more likely to block attacks.

E-swords were never very good.

>tweak: Bear traps now do slightly less than twice as much damage.

These things would only occasionally break your bones. These aren't rubberized, "humane" 'starve-traps'. These are for breaking legs.

>tweak: Null-Rod isn't ridiculously deadly anymore. Thank God you never found out

It used to have an insane 15 force.

>tweak: Pulse weapons have been nerfed. They are less accurate, take longer to shoot, and have a much higher chance to miss due to the usage of more, much weaker projectiles. They lock the wielder in place for a significant period now. Now use unique sprites for their projectiles, shooting pulses rather than lasers. No longer hitscan.

Pulse weapons are really fucking deadly. They didn't have a downside. Also they shoot pulses now, instead of multiple lasers.

>tweak: Uzi has had all values for unwielded inaccuracy shifted downwards by 1.

It's a machine pistol. You should be able to dual-wield it without a massive detriment.. It's not like this thing is a fucking P90.

>tweak: Tasers are no longer hitscan weapons. Shock beam is unaffected.

Creates tension by allowing the possiblity to weave out of the way from the electrode.

>tweak: shock beam has a unique sound.

It sounds cool.

>tweak: Energy Crossbow has some armor penetration and a minor EMP.

It was literally useless against armor. ANY armor would fully negate the damage.  It'd also discharge guns this way.


>tweak: Phoron pistol does 50% as much damage, and uses its own sound.

Toxins damage is terrifying on Zuhmed.

chaoko99 commented:
https://baystation12.net/forums/threads/make-tasers-no-longer-hitscan.5685/#post-83013

chaoko99 committed:
Shitshit
Makes the Electric chair more brutal.

Makes the null rod less insane.

Bear trap is kilier.

Adds new sounds and behaviors for guns.

Weapons are killier.

blazerules commented:
I don't think you understand what a replica is if you think it should do less damage. Hell... todays replicas/reproductions would actually be much, much better than original katanas simply due to improved steel quality so...

Like... you do know a replica is simply a copy of something made by the original maker? So a guy makes a sword. Then he makes the exact same sword. That sword would be a replica of the first. There is nothing in the universe saying that making the exact same sword would make it somehow worse. Do you even sword?

Please do tell me how getting hit with a hardened steel sharp blade wouldn't hurt at all (Spoiler it would probably fuck you up)

chaoko99 commented:
I figured it was like a poor quality mall katana or something. Unsharpened. I've been thinking about adding sharpening stones anyways.. Maybe a grinder tool. I dunno.

Techhead0 commented on code/game/objects/items/weapons/material/swords.dm:
```diff
@@ -36,5 +36,5 @@
-	force_divisor = 0.2
-	thrown_force_divisor = 0.2
+	force_divisor = 0.1
+	thrown_force_divisor = 0.1
```
Not a fan of reducing it this much, it's still a long metal stick. Somewhere like `0.15` sound okay?

Techhead0 commented on code/game/objects/items/weapons/melee/energy.dm:
```diff
@@ -6,6 +6,11 @@
+	var/attack_sound = 'sound/weapons/blade1.ogg' //Snowflakey, but adding unique sounds to everything south of /item/ would be a large undertaking.
+
+/obj/item/weapon/melee/energy/attack(var/mob/living/M, var/mob/living/user)
+	playsound(get_turf(user), 50, 1)
```
Shouldn't this... you know... specify the sound?

Techhead0 commented on code/game/objects/items/weapons/melee/energy.dm:
```diff
@@ -143,14 +145,20 @@
 
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if(H.flags & NO_BLOCK)
```
Should be `H.species.flags`

Techhead0 commented on code/game/objects/items/weapons/material/twohanded.dm:
```diff
@@ -107,15 +107,15 @@
 	// 12/19 with hardness 60 (steel) or 10/16 with hardness 50 (glass)
 	force_divisor = 0.33
-	unwielded_force_divisor = 0.20
+	unwielded_force_divisor = 0.66
```
You made one-handed spear use better than two-handed?

chaoko99 replied:
Crap. I figured the UFD was calculated after the base Fd.

Techhead0 reviewed (commented):
Only hit the melee stuff, will be back for the ranged stuff.

Techhead0 commented on code/modules/projectiles/guns/energy/pulse.dm:
```diff
@@ -6,15 +6,16 @@
-	burst_delay = 3
-	burst = 3
+	burst_delay = 1.5
+	burst = 9
```
Large burst sizes with light damage is problematic because light wounds are very likely to autoheal, especially if they land on separate organs. 8 damage is okay, but 6 and 4 on the smaller pulse weapons will have a propensity to tickle. Instead of 9 shots at 8 damage, I'd do something 6 shots at 10, and simply do shorter bursts on the smaller guns.

chaoko99 replied:
That does less damage than it would've before.  The damage is lower (in increments) to account for the fact that the pulses can hit internal organs and just kill you like that. It rolls a lot of chances for that. Atop this, it's still a lot of pain damage which does a good enough job.  sane and mature dictates that people should not like being hurt. Atop this, the whole point of making long weak bursts a thing is to make it so the weapon locks the firer in place for an extended period—Forcing them to commit to their attack.

chaoko99 replied:
Opinion on this?

Techhead0 replied:
Like I said, 8 is okayish, but I think 4 or 6 damage is laughable.

Also... non-integer `burst_delay` doesn't work because non-integer `sleep()` doesn't do anything. `sleep(1.5)` only sleeps for one tick.

Techhead0 commented on code/modules/projectiles/projectile/beams.dm:
```diff
@@ -61,19 +61,20 @@
 	fire_sound='sound/weapons/pulse.ogg'
-	damage = 15 //lower damage, but fires in bursts
+	damage = 4 //lower damage, but fires in bursts
+	hitscan = 0
```
If non-hitscan with non-beam sprites, should probably be `obj/item/projectile/energy` subtype.

chaoko99 replied:
Projectiles embed.

chaoko99 replied:
er. scratch that.

Techhead0 commented on code/modules/projectiles/projectile/beams.dm:
```diff
@@ -26,17 +26,17 @@
 
 /obj/item/projectile/beam/smalllaser
-	damage = 25
+	damage = 23
```
Pointless change?

Techhead0 reviewed (commented):
Here's part 2, after some delay. I didn't comment on stuff other people already pointed out.

chaoko99 commented:
Fuuuuck this.

chaoko99 commented:
wait no

chaoko99 committed:
Update pulse.dm

chaoko99 committed:
Update pulse.dm but this time it works.

chaoko99 committed:
Update beams.dm

chaoko99 committed:
Update energy.dm

chaoko99 committed:
Update energy.dm

chaoko99 committed:
Update pulse.dm

chaoko99 committed:
Update energy.dm

Techhead0 commented on code/game/objects/items/weapons/material/twohanded.dm:
```diff
@@ -107,15 +107,15 @@
-	force_divisor = 0.33
-	unwielded_force_divisor = 0.20
+	force_divisor = 0.66
+	unwielded_force_divisor = 0.66
```
A) Two-handing a spear should at least give you *some* bonus.
B) ~~40~~ 39 force for one-handed use is a little crazy.
C) Update the comment.

Techhead0 reviewed (commented):
Needs a squash, also has conflicts to be resolved.

chaoko99 commented:
But I'm not sure how the math woooorks and I don't wanna be talked down to
by Zuhayr for askiiiing...!

On Fri, Jan 12, 2018 at 7:10 PM, Techhead <notifications@github.com> wrote:

> *@Techhead0* commented on this pull request.
>
> Needs a squash, also has conflicts to be resolved.
> ------------------------------
>
> In code/game/objects/items/weapons/material/twohanded.dm
> <https://github.com/Baystation12/Baystation12/pull/19688#discussion_r161361722>
> :
>
> >  	w_class = ITEM_SIZE_HUGE
>  	slot_flags = SLOT_BACK
>
>  	// 12/19 with hardness 60 (steel) or 10/16 with hardness 50 (glass)
> -	force_divisor = 0.33
> -	unwielded_force_divisor = 0.20
> +	force_divisor = 0.66
> +	unwielded_force_divisor = 0.66
>
> A) Two-handing a spear should at least give you *some* bonus.
> B) 40 39 force for one-handed use is a little crazy.
> C) Update the comment.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/Baystation12/Baystation12/pull/19688#pullrequestreview-88646800>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEfPBLtO8HqrNw5dFbuoQ-yQ5JdnkcDPks5tKB6VgaJpZM4Q51vf>
> .
>

Techhead0 commented:
Math is divisor*hardness or divisor*weight, depending on what it's calculating.

Thrown objects and blunt weapons use weight, pointy weapons use hardness.

(Also, yes I know it says "divisor" when it should be "factor")

Spookerton commented:
I'm basically against every single change in this PR, lol. 👎

chaoko99 commented:
I mean. If I *really* start caring, I might just tear the visual glitter out of this PR and use that.

chaoko99 commented:
Closing this out so I can scrub out all the bad.
